### PR TITLE
Automatically check for SPDX headers and linting

### LINKED ISSUE #26: Enforce SPDX check
SPDX should be automatically enforced in Github Actions.

### PR DESCRIPTION
**Pull request by mkannwischer**
Resolves #26

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -13,4 +13,4 @@
 
 * [ ] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
 
-<!-- Once your pull request is ready for review and passing continuous integration tests, please convert from a draft PR to a normal PR, and request a review from one of the OQS core team members. -->
\ No newline at end of file
+<!-- Once your pull request is ready for review and passing continuous integration tests, please convert from a draft PR to a normal PR, and request a review from one of the OQS core team members. -->
```
Comment (hanno-becker) [inline .github/pull_request_template.md:16]: `OQS core team members` -> `mlkem-c-aarch64 maintainers`?

Comment (mkannwischer) [reply .github/pull_request_template.md:16]: I think that is unrelated to my changes? 
We need to clean up:
- https://github.com/pq-code-package/mlkem-c-aarch64/blob/main/.github/ISSUE_TEMPLATE/bug_report.md
- https://github.com/pq-code-package/mlkem-c-aarch64/blob/main/.github/ISSUE_TEMPLATE/feature_request.md
- https://github.com/pq-code-package/mlkem-c-aarch64/blob/main/CONTRIBUTING.md
- https://github.com/pq-code-package/mlkem-c-aarch64/blob/main/GOVERNANCE.md
- https://github.com/pq-code-package/mlkem-c-aarch64/blob/main/SECURITY.md
- https://github.com/pq-code-package/mlkem-c-aarch64/blob/main/SUPPORT.md

File: scripts/format
Comment (hanno-becker) [inline scripts/format:1]: When is this script (supposed to be) used?

Comment (mkannwischer) [reply scripts/format:1]: The developer can use this to format the code locally before pushing.

Comment (hanno-becker) [reply scripts/format:1]: Ok

File: CODEOWNERS [hunk 2]
```diff
@@ -5,4 +5,4 @@
 
 # Default owner
 * @pq-code-package/pqcp-hackers
```
Comment (hanno-becker) [inline CODEOWNERS:7]: github complains:
```
Unknown owner on line 7: make sure the team @pq-code-package/pqcp-hackers exists, is publicly visible, and has write access to the repository
* @pq-code-package/pqcp-hackers
```

File: .github/pull_request_template.md (hunk 1 above)
Comment (mkannwischer) [reply .github/pull_request_template.md:16]: Let's discuss in a seperate PR: https://github.com/pq-code-package/mlkem-c-aarch64/pull/41

File: CODEOWNERS (hunk 2 above)
Comment (mkannwischer) [reply CODEOWNERS:7]: I fixed this in https://github.com/pq-code-package/mlkem-c-aarch64/pull/41

**Review by hanno-becker — APPROVED** [review]

**Pull request merged**